libs/sdk-model/src/insight/sanitization.ts

```diff
@@ -1,32 +1,62 @@
 // (C) 2020 GoodData Corporation
 import { InvariantError } from "ts-invariant";
+import { uniqWith } from "lodash";
 
-import { insightSorts, IInsightDefinition, insightBuckets, insightSetBuckets } from ".";
-import { bucketTotals, IBucket, bucketAttributeIndex, bucketSetTotals } from "../execution/buckets";
-import { ISortItem, isAttributeSort, sortEntityIds, isMeasureSort } from "../execution/base/sort";
+import { IInsightDefinition, insightAttributes, insightBuckets, insightSetBuckets, insightSorts } from ".";
+import { bucketAttributeIndex, bucketSetTotals, bucketTotals, IBucket } from "../execution/buckets";
+import { isAttributeSort, isMeasureSort, ISortItem, sortEntityIds } from "../execution/base/sort";
 import { ITotal } from "../execution/base/totals";
-
+import { attributeLocalId, isAttribute } from "../execution/attribute";
+import { areObjRefsEqual } from "../objRef";
 /**
  * Makes sure the insight does not have any nonsensical data (like totals that no longer make sense, etc.), before it is saved.
  *
  * @param insight - the insight or insight definition to sanitize
  * @public
  */
 export function insightSanitize<T extends IInsightDefinition>(insight: T): T {
-    return removeInvalidTotals(insight);
+    return removeInvalidTotals(removeDuplicateDrillAttributes(insight));
+}
+
+function removeDuplicateDrillAttributes<T extends IInsightDefinition>(insight: T): T {
+    const removed = insightBuckets(insight).map((bucket) => {
+        if (bucket.localIdentifier === "attribute") {
+            bucket.items = uniqWith(bucket.items, (firstItem, secondItem) => {
+                if (isAttribute(firstItem) && isAttribute(secondItem)) {
+                    return areObjRefsEqual(firstItem.attribute.displayForm, secondItem.attribute.displayForm);
+                }
+
+                return false;
+            });
```

Comment:
is this mutation needed? perhaps do newBucket(oldId, newItems)?

Reply:
:+1: added newBucket